src/headers.zig

```diff
@@ -38,20 +33,17 @@ pub const Headers = struct {
     }
 
     pub fn appendValue(self: *Headers, name: []const u8, value: []const u8) !void {
-        const duped_name = try self.allocator.dupe(u8, name);
-        const duped_value = try self.allocator.dupe(u8, value);
-
-        try self.list.append(.{
-            .name = duped_name,
-            .value = duped_value,
+        try self.append(.{
+            .name = name,
+            .value = value,
         });
     }
 
-    pub inline fn append(self: *Headers, header: Header) !void {
-        return self.appendValue(header.name, header.value);
+    pub fn append(self: *Headers, header: Header) !void {
```

Comment:
It is very likely that this header was obtained from a hzzp event (and therefore is nearly guaranteed to point to invalid memory before the Headers object is deinitialized) or from another Headers object, which may or may not outlive this object. This should be noted.